src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java

```diff
@@ -520,6 +530,266 @@ public void xmlRouteBuilder_RestReportIdWorkloadInventory_IdParamGiven_ShouldCal
+        workloadInventoryReportModels.add(wir2);
+
+        WorkloadInventoryReportModel wir3 = new WorkloadInventoryReportModel();
+        wir3.setProvider("provider");
+        wir3.setDatacenter("datacenter");
+        workloadInventoryReportModels.add(wir3);
+
+        analysisService.addWorkloadInventoryReportModels(workloadInventoryReportModels, analysisModel.getId());
+
+
+        Map<String, Object> variables = new HashMap<>();
+        variables.put("id", analysisModel.getId());
+
+        String orderBy = "provider";
+        variables.put("orderBy", orderBy);
+        Boolean orderAsc = true;
+        variables.put("orderAsc", orderAsc);
+
+        String datacenter1 = "my datacenter1";
+        variables.put("datacenter1", datacenter1);
+        String datacenter2 = "my datacenter2";
+        variables.put("datacenter2", datacenter2);
+
+        StringBuilder sb = new StringBuilder("")
+                .append("orderBy={orderBy}&")
+                .append("orderAsc={orderAsc}&")
+                .append("datacenter={datacenter1}&")
+                .append("datacenter={datacenter2}");
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(TestUtil.HEADER_RH_IDENTITY, TestUtil.getBase64RHIdentity());
+        HttpEntity<String> entity = new HttpEntity<>(null, headers);
+
+        //When
+        camelContext.start();
+        TestUtil.startUsernameRoutes(camelContext);
+        camelContext.startRoute("to-sortBean");
+        camelContext.startRoute("to-workloadInventoryFilterBean");
+        camelContext.startRoute("filtered-workload-inventory-report-get-details-as-csv");
+        camelContext.startRoute("workload-inventory-report-model-to-csv");
+
+        ResponseEntity<String> response = restTemplate.exchange(camel_context + "report/{id}/workload-inventory/filtered-csv?" + sb.toString(), HttpMethod.GET, entity, String.class, variables);
+
+        //Then
+        SortBean sortBean = new SortBean(orderBy, orderAsc);
+
+        WorkloadInventoryFilterBean filterBean = new WorkloadInventoryFilterBean();
+        filterBean.setDatacenters(new HashSet<>(Arrays.asList(datacenter1, datacenter2)));
+
+        verify(workloadInventoryReportService).findByAnalysisOwnerAndAnalysisId(analysisModel.getOwner(), analysisModel.getId(), sortBean, filterBean);
+        Assert.assertTrue(response.getHeaders().get("Content-Type").contains("text/csv"));
+        Assert.assertTrue(response.getHeaders().get("Content-Disposition").contains("attachment;filename=workloadInventory_1.csv"));
+        assertThat(response).isNotNull();
+        assertThat(response.getBody()).contains("Provider,Datacenter,Cluster,VM name,OS type,Operating system description,Disk space,Memory,CPU cores,Workload,Effort,Recommended targets,Flags IMS,Product,Version,HostName");
+        assertThat(response.getBody()).contains("my providerA,my datacenter2");
+        assertThat(response.getBody()).contains("my providerB,my datacenter1");
```

Comment:
I assume your goal was to check that "provider" is not on the list, but you are only checking the ones thar are in the list but not the not which should not be in the list.
Am I wrong ?

Reply:
@jonathanvila Thanks for your review, my last commit added a new assert to make sure we have only 3 rows (including the headers) so we know there are no extra rows